### PR TITLE
jobs: add pct job details struct

### PR DESCRIPTION
**Pull request by MaikoTan**
Didn't test, i don't know about OP....

EDIT: I try be best, created a PR at the OP side: https://github.com/OverlayPlugin/OverlayPlugin/pull/372

### CODE REVIEW

File: ui/test/test.ts [hunk 1]
```diff
@@ -114,6 +114,19 @@ addOverlayListener('onPlayerChangedEvent', (e) => {
     } else if (detail.job === 'RPR' && detail.jobDetail) {
       jobInfo.innerText =
         `${detail.jobDetail.soul} | ${detail.jobDetail.shroud} | ${detail.jobDetail.enshroudMilliseconds} | ${detail.jobDetail.lemureShroud} | ${detail.jobDetail.voidShroud}`;
+    } else if (detail.job === 'PCT' && detail.jobDetail) {
+      jobInfo.innerText =
+        `${detail.jobDetail.palleteGauge} | ${detail.jobDetail.palleteGauge} | (${detail.jobDetail.creatureMotif}|${
+          detail.jobDetail.weaponMotif ? 'Weapon' : 'None'
+        }|${detail.jobDetail.landscapeMotif ? 'Landscape' : 'None'}) | (${
+          detail.jobDetail.depictions.join('|') || 'None'
+        }) | ${
```
Comment (wexxlee) [inline ui/test/test.ts:123]:
```suggestion
        `${detail.jobDetail.palleteGauge} | ${detail.jobDetail.paint} | (${detail.jobDetail.creatureMotif} | ${
          detail.jobDetail.weaponMotif ? 'Weapon' : 'None'
        } | ${detail.jobDetail.landscapeMotif ? 'Landscape' : 'None'}) | (${
          detail.jobDetail.depictions.join('+') || 'None'
        }) | ${
```

Comment (wexxlee) [reply ui/test/test.ts:123]: Some minor formatting adjustments & prop change.
Also, purely stylistic suggestion so feel free to ignore - but perhaps consider joining the depictions with a `+` instead to distinguish from the field separator, if you think that makes sense?

**Review by wexxlee — CHANGES_REQUESTED** [review]
Thanks for this!  I'm working on other stuff before I get PCT fully leveled, so I won't be able to test this for a bit either.  But the structs look right to me.

Minor formatting/prop change requested, and a purely optional stylistic suggestion (take it or leave it as you see fit).

File: ui/test/test.ts (hunk 1 above)
Comment (MaikoTan) [reply ui/test/test.ts:123]: This looks make sense to me, thanks for the fixing.

File: plugin/CactbotEventSource/FFXIVProcessIntl.cs [hunk 2]
```diff
@@ -840,5 +842,71 @@ public struct ReaperJobMemory {
...
+          Landscape = 1 << 5,
+      }
+
+      [Flags]
+      private enum CreatureFlags : byte {
+          Pom = 1,
+          Wings = 1 << 1,
+          Claw = 1 << 2,
+          // Maw = 1 << 3, // Once you paint the Maw motif, it becomes a Madeen portrait.
+          MooglePortait = 1 << 4,
```
Comment (MaikoTan) [inline plugin/CactbotEventSource/FFXIVProcessIntl.cs:864]:
```suggestion
          MooglePortrait = 1 << 4,
```

File: plugin/CactbotEventSource/FFXIVProcessIntl.cs [hunk 3]
```diff
@@ -840,5 +842,71 @@ public struct ReaperJobMemory {
...
+            motifs.Add("Pom");
+          if (creatureFlags.HasFlag(CanvasFlags.Wing))
+            motifs.Add("Wing");
+          if (creatureFlags.HasFlag(CanvasFlags.Claw))
+            motifs.Add("Claw");
+          return motifs.ToArray();
+        }
+      }
+
+      public bool mooglePortrait => creatureFlags.HasFlag(CreatureFlags.MooglePortait);
```
Comment (MaikoTan) [inline plugin/CactbotEventSource/FFXIVProcessIntl.cs:908]:
```suggestion
      public bool mooglePortrait => creatureFlags.HasFlag(CreatureFlags.MooglePortrait);
```

**Review by wexxlee — APPROVED** [review]

**Pull request merged**